### PR TITLE
Pass ConnectorSession to getSplits in JDBC connector

### PR DESCRIPTION
**Pull request by wendigo**

### CODE REVIEW

File: presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/BaseJdbcClient.java [hunk 1]
```diff
@@ -339,7 +339,7 @@ protected static ResultSet getColumns(JdbcTableHandle tableHandle, DatabaseMetaD
     }
 
     @Override
-    public ConnectorSplitSource getSplits(JdbcIdentity identity, JdbcTableHandle tableHandle)
+    public ConnectorSplitSource getSplits(ConnectorSession session, JdbcIdentity identity, JdbcTableHandle tableHandle)
```
Comment (findepi) [inline presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/BaseJdbcClient.java]: Then you should remove `identity` parameter. it's part of the session

Comment (wendigo) [reply presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/BaseJdbcClient.java]: Great :) One parameter less

**Comment by electrum** [comment]
I'm wondering if we should get rid of `JdbcIdentity` entirely. It seems somewhat arbitrary now as to what methods use it.

**Comment by findepi** [comment]
`JdbcIdentity` is used as a cache key (eg in CachingJdbcClient). `ConnectorSession` cannot be used for this purpose.
Anyway, this isn't (and should be) part of this PR, so I will merge it as is.
Thanks @electrum for review!

**Pull request merged**